lib/eex/lib/eex/compiler.ex

```diff
@@ -84,27 +97,39 @@ defmodule EEx.Compiler do
         "please remove \"#{modifier}\" accordingly"
 
     :elixir_errors.erl_warn(line, state.file, message)
-    generate_buffer([{:middle_expr, line, '', chars, trimmed?} | t], buffer, scope, state)
+    generate_buffer([{:middle_expr, line, column, '', chars, trimmed?} | t], buffer, scope, state)
     # TODO: Make this an error on Elixir v2.0 since it accidentally worked previously.
     # raise EEx.SyntaxError, message: message, file: state.file, line: line
   end
 
-  defp generate_buffer([{:middle_expr, line, _, chars, _} | _], _buffer, [], state) do
+  defp generate_buffer([{:middle_expr, line, _column, _, chars, _} | _], _buffer, [], state) do
```

Comment:
In the future column could be used to do even better error messages.

Reply:
Should we use the opportunity and add those now to EEx.SyntaxError? Is should be a small amount of work.